src/inc/msquic.hpp

```diff
@@ -207,20 +207,30 @@ class CxPlatThread {
 
 #ifdef CXPLAT_FRE_ASSERT
 
+#ifndef _KERNEL_MODE
+#include <stdio.h> // For printf below
+#endif
+
 class CxPlatWatchdog {
     CxPlatEvent ShutdownEvent {true};
     CxPlatThread WatchdogThread;
     uint32_t TimeoutMs;
+    bool WriteToConsole;
     static CXPLAT_THREAD_CALLBACK(WatchdogThreadCallback, Context) {
         auto This = (CxPlatWatchdog*)Context;
         if (!This->ShutdownEvent.WaitTimeout(This->TimeoutMs)) {
+#ifndef _KERNEL_MODE // Not supported in kernel mode
+            if (This->WriteToConsole) {
+                printf("Error: Watchdog timeout fired!\n");
```

Comment:
If error, how about `fprintf(stderr, )` ?

Reply:
Printing to error stream causes other complexity at the PowerShell level if you decide you want to ignore certain errors (temporarily) like we are doing currently. So I prefer to keep it this way.